cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/SecureStoreHandler.java

```diff
@@ -78,54 +68,43 @@ public void create(HttpRequest httpRequest, HttpResponder httpResponder, @PathPa
                                       " \n" + GSON.toJson(dummy));
     }
 
-    String description = secureKeyCreateRequest.getDescription();
-    String value = secureKeyCreateRequest.getData();
-    if (Strings.isNullOrEmpty(value)) {
-      throw new BadRequestException("The data field should not be empty. This is the data that will be stored " +
-                                      "securely.");
-    }
-
-    byte[] data = value.getBytes(StandardCharsets.UTF_8);
-    secureStoreManager.put(namespace, name, data, description, secureKeyCreateRequest.getProperties());
+    secureStoreService.put(secureKeyCreateRequest, secureKeyId);
     httpResponder.sendStatus(HttpResponseStatus.OK);
   }
 
   @Path("/keys/{key-name}")
   @DELETE
   public void delete(HttpRequest httpRequest, HttpResponder httpResponder, @PathParam("namespace-id") String namespace,
-                     @PathParam("key-name") String name) throws IOException {
-    secureStoreManager.delete(namespace, name);
+                     @PathParam("key-name") String name) throws Exception {
+    SecureKeyId secureKeyId = new SecureKeyId(namespace, name);
+    secureStoreService.delete(secureKeyId);
     httpResponder.sendStatus(HttpResponseStatus.OK);
   }
 
   @Path("/keys/{key-name}")
   @GET
   public void get(HttpRequest httpRequest, HttpResponder httpResponder, @PathParam("namespace-id") String namespace,
-                  @PathParam("key-name") String name) throws IOException {
-    SecureStoreData secureStoreData = secureStore.get(namespace, name);
-    String data = new String(secureStoreData.get(), StandardCharsets.UTF_8);
+                  @PathParam("key-name") String name) throws Exception {
+    SecureKeyId secureKeyId = new SecureKeyId(namespace, name);
+    String data = new String(secureStoreService.get(secureKeyId).get(), StandardCharsets.UTF_8);
```

Comment:
Yes, thats what we want. Otherwise is some one changes their default charset after adding some keys, the whole thing will break.


> On Jul 22, 2016, at 2:25 PM, Bhooshan Mogal <notifications@github.com> wrote:
> 
> In cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/SecureStoreHandler.java <https://github.com/caskdata/cdap/pull/6134#discussion_r71948484>:
> 
> >      httpResponder.sendStatus(HttpResponseStatus.OK);
> >    }
> >  
> >    @Path("/keys/{key-name}")
> >    @GET
> >    public void get(HttpRequest httpRequest, HttpResponder httpResponder, @PathParam("namespace-id") String namespace,
> > -                  @PathParam("key-name") String name) throws IOException {
> > -    SecureStoreData secureStoreData = secureStore.get(namespace, name);
> > -    String data = new String(secureStoreData.get(), StandardCharsets.UTF_8);
> > +                  @PathParam("key-name") String name) throws Exception {
> > +    SecureKeyId secureKeyId = new SecureKeyId(namespace, name);
> > +    String data = new String(secureStoreService.get(secureKeyId).get(), StandardCharsets.UTF_8);
> but you're hardcoding the charset here?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/caskdata/cdap/pull/6134/files/5d0bbcef12f2e4cf3e806f96254510dc0d77d2e2#r71948484>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AAFGk2Sh2JzC-U2CnqM2jKJS4INpVZx2ks5qYTVMgaJpZM4JSQa6>.
>